addons/binding/org.openhab.binding.blebox/src/main/java/org/openhab/binding/blebox/internal/discovery/BleboxScanner.java

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.blebox.internal.discovery;
+
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import org.apache.commons.net.util.SubnetUtils;
+import org.apache.commons.net.util.SubnetUtils.SubnetInfo;
+import org.apache.http.client.fluent.Async;
+import org.apache.http.client.fluent.Content;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.concurrent.FutureCallback;
+import org.openhab.binding.blebox.devices.DeviceInfo;
+import org.openhab.binding.blebox.devices.StatusResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link BleboxScanner} is responsible for scan local network for Blebox devices
+ *
+ * @author Szymon Tokarski - Initial contribution
+ */
+public class BleboxScanner {
+    private Logger logger = LoggerFactory.getLogger(BleboxScanner.class);
+
+    static final int ENVISALINK_BRIDGE_PORT = 4025;
```

Comment:
If you copy-paste code (EnvisalinkBridgeDiscovery), at least make sure to remove unused code. And maybe it is nice to add the original author's name here as well.